src/entt/meta/meta.hpp

```diff
@@ -1464,6 +1462,66 @@ class meta_type {
 }
 
 
+[[nodiscard]] inline meta_any meta_func::invoke (meta_handle instance, meta_any* const args, const std::size_t sz) const {
```

Comment:
I thought about this change. Imho, `meta_func` isn't the right place for it.

In fact, `meta_func` is a wrapper for a function node. I don't expect it to keep iterating the list of available functions of its type for some reasons. It refers to a node, I expect it to use this node and eventually to fail if I do something wrong (like, passing to it the arguments for a different overload).
This is as if you had a pointer to member for an overloaded function, you passed it the list of arguments for another overload and the language or the compiler decided that the pointer to member function is somehow _resolved_ to something different for you under the hood. A little misleading.

My opinion is that we have now these functions in for the meta type:
```
[[nodiscard]] meta_range<meta_func> func() const ENTT_NOEXCEPT;
[[nodiscard]] meta_func func(const id_type id) const;
```
The second one is redundant, you can reach the same result with a range and a manual search.
On the other side, we could rework the second function as:
```
template<typename... Args>
[[nodiscard]] meta_any invoke(const id_type id, Args &&... args) const;
```
Then, we can resolve the overload during name/params resolution.

If you think a moment to it, `meta_type::construct` does exactly the same thing.
Similarly, we could remove the second overload for `data` and introduce a pair `set`/`get` directly on the meta type. And so on for the other meta objects.

This would make using the meta type _easier_too. From:
```
auto res = type.func("foo"_hs).invoke({}, 42);
```
To:
```
auto res = type.invoke("foo"_hs, {}, 42);
```

What are your thoughts on this?
I don't expect you to do that, don't worry. I can easily rework the meta type and prepare the whole thing for the function overload. There shouldn't even be conflicts.

Reply:
For future generations: the shortcuts described here implemented in #552